src/common/com/intellij/plugins/haxe/lang/psi/impl/AbstractHaxePsiClass.java

```diff
@@ -528,4 +528,18 @@ public HaxeComponentName getComponentName() {
     }
   };
 
+
+  // FIX: for twice deletion of file in project view (issue #424)
+  @Override
+  public void delete() {
+    final HaxeFile file = (HaxeFile)getContainingFile();
+    if(file != null) {
+      final PsiClass[] classes = file.getClasses();
+      if(classes.length == 1 && classes[0] == this) {
+        file.delete();
+        return;
+      }
+    }
+    super.delete();
```

Comment:
@eliasku Do you think that the file deletion triggers is as clean as deleting the class and then the file?  My thinking is that the super.delete() fixes up the PSI tree, and triggers index updates (e.g. FindUsages, Hierarchies).  That said, I haven't run this code myself to verify whether all of the correct things are triggered.

I would probably do this:
  super.delete();
  classes = file.getClasses();
  if (len(classes) == 0) {
    file.delete();
  }

But I'm not you.  ;)